### PR TITLE
Collect metadata in ipi-deprovision

### PR DESCRIPTION
**Pull request by hongkailiu**

### CODE REVIEW

**Comment by hongkailiu** [comment]
/hold
need to inject the secret for `oc login` before merging this.

File: cmd/ipi-deprovision/ipi-deprovision.sh [hunk 1]
```diff
@@ -29,7 +29,45 @@ handle_cluster () {
...
+        continue
+      fi
+      echo "oc get pod -n ${ns} ${pod_name} -o yaml"
+      oc get pod -n ${ns} ${pod_name} -o yaml
+      echo "oc describe pod -n ${ns} ${pod_name}"
+      oc describe pod -n ${ns} ${pod_name}
+      for container_name in $(oc get pod -n ci-op-zf3n7jgn  e2e-aws-serial -o=jsonpath='{.spec.containers[*].name}')
+      do
+        echo "oc logs -n ${ns} ${pod_name} -c ${container_name}"
+        oc logs -n ${ns} ${pod_name} -c ${container_name}
```
Comment (stevekuznetsov) [inline cmd/ipi-deprovision/ipi-deprovision.sh]: We have the logs in GCS already from the job -- we want to `describe` to get events for the Pod

File: cmd/ipi-deprovision/ipi-deprovision.sh [hunk 2]
```diff
@@ -29,7 +29,45 @@ handle_cluster () {
...
+collect_metadata () {
+  local cluster_name
+  cluster_name=$1
+  ###eg cluster_name=kubernetes.io/cluster/ci-op-724qy8fn-55c01-s5c8w
+  local ns
+  ns=${cluster_name:22:14}
+  echo "cluster is used by namespace ${ns}"
+  echo "oc get pod -n ${ns}"
+  oc get pod -n ${ns}
+  for pod_name in $(oc get pod -n $ns -o=jsonpath='{.items[*].metadata.name}')
```
[the remark this answers is not in the document]
Comment (hongkailiu) [reply cmd/ipi-deprovision/ipi-deprovision.sh]: why cannot I see this coming ... will do tonight.

File: cmd/ipi-deprovision/ipi-deprovision.sh [hunk 3]
```diff
@@ -29,7 +29,45 @@ handle_cluster () {
...
 
-
+collect_metadata () {
+  local cluster_name
+  cluster_name=$1
+  ###eg cluster_name=kubernetes.io/cluster/ci-op-724qy8fn-55c01-s5c8w
+  local ns
+  ns=${cluster_name:22:14}
+  echo "cluster is used by namespace ${ns}"
+  echo "oc get pod -n ${ns}"
```
[the remark this answers is not in the document]
Comment (hongkailiu) [reply cmd/ipi-deprovision/ipi-deprovision.sh]: one of my bad choices.

File: cmd/ipi-deprovision/ipi-deprovision.sh (hunk 1 above)
Comment (hongkailiu) [reply cmd/ipi-deprovision/ipi-deprovision.sh]: Will remove. describe is done above.

File: cmd/ipi-deprovision/ipi-deprovision.sh [hunk 4]
```diff
@@ -29,7 +29,45 @@ handle_cluster () {
...
+  oc get pod -n ${ns}
+  for pod_name in $(oc get pod -n $ns -o=jsonpath='{.items[*].metadata.name}')
+  do
+    echo "inpecting ${pod_name} ..."
+    local env_cluster_name_value
+    ###eg env_cluster_name_value=ci-op-724qy8fn-55c01
+    env_cluster_name_value=$(oc get pod -n ${ns} ${pod_name} -o json | jq -r '.spec.containers[] | select (.name == "setup") | .env[] | select (.name == "CLUSTER_NAME") | .value')
```
[the remark this answers is not in the document]
Comment (hongkailiu) [reply cmd/ipi-deprovision/ipi-deprovision.sh]: Will do tonight.

File: cmd/ipi-deprovision/ipi-deprovision.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (hongkailiu) [reply cmd/ipi-deprovision/ipi-deprovision.sh]: Done.

File: cmd/ipi-deprovision/ipi-deprovision.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (hongkailiu) [reply cmd/ipi-deprovision/ipi-deprovision.sh]: Done.

File: cmd/ipi-deprovision/ipi-deprovision.sh (hunk 4 above)
[the remark this answers is not in the document]
Comment (hongkailiu) [reply cmd/ipi-deprovision/ipi-deprovision.sh]: Comparing the last version, we lost the following checking (I did not find a way to do it inside `jq`)

```
if [[ ! ${cluster_name} == *"${env_cluster_name_value}"* ]]; then	
        echo "${env_cluster_name_value} is not contained in ${cluster_name}!"	
        continue	
fi
```

But I guess that is fine.

File: cmd/ipi-deprovision/ipi-deprovision.sh (hunk 1 above)
Comment (hongkailiu) [reply cmd/ipi-deprovision/ipi-deprovision.sh]: Done.

File: cmd/ipi-deprovision/ipi-deprovision.sh [hunk 5]
```diff
@@ -39,6 +56,9 @@ do
   do
     expirationDateValue=$(echo ${json_output} | jq --arg cl "${cluster}" -r -S '.Vpcs[] | select (.Tags[]? | (.Key == $cl and .Value == "owned")) | .Tags[] | select (.Key == "expirationDate") | .Value')
     handle_cluster "${cluster}" "${expirationDateValue}" "${r}"
+    echo "oc status"
+    oc status
```
Comment (stevekuznetsov) [inline cmd/ipi-deprovision/ipi-deprovision.sh]: What will this print for us? status of which namespace, for every cluster?

File: cmd/ipi-deprovision/ipi-deprovision.sh [hunk 6]
```diff
@@ -29,7 +29,24 @@ handle_cluster () {
...
+  cluster_name=$1
+  ###eg cluster_name=kubernetes.io/cluster/ci-op-724qy8fn-55c01-s5c8w
+  local ns
+  ns=${cluster_name:22:14}
+  echo "cluster is used by namespace ${ns}"
+  echo "oc get pod -n ${ns}"
+  local cache_json
+  cache_json=$(oc get pods -n "${ns}" -o json)
+  ###eg, the value of env var CLUSTER_NAME: ci-op-724qy8fn-55c01
+  for pod_name in $(echo "${cache_json}" | jq --arg prefix "${ns}-" -r '.items[] | . as $pod | $pod.spec.containers[] | select(.name == "setup") | .env[]? | select(.name == "CLUSTER_NAME" and (.value | startswith($prefix) )) | $pod.metadata.name')
```
Comment (stevekuznetsov) [inline cmd/ipi-deprovision/ipi-deprovision.sh]: Why check the prefix when we know exactly the value of the cluster name to check?

File: cmd/ipi-deprovision/ipi-deprovision.sh (hunk 5 above)
Comment (hongkailiu) [reply cmd/ipi-deprovision/ipi-deprovision.sh]: Removed.
I was expecting to use this to confirm that oc-login was set up ok.

File: cmd/ipi-deprovision/ipi-deprovision.sh (hunk 6 above)
Comment (hongkailiu) [reply cmd/ipi-deprovision/ipi-deprovision.sh]: Totally true.

**Comment by wking** [comment]
Out of curiousity, what was the purpose of this collection?

**Pull request merged**